src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/integration/wiremock/Elite2MockServer.kt

```diff
@@ -104,12 +104,29 @@ class Elite2MockServer : WireMockRule(8999) {
     stubFor(get(urlEqualTo("/api/agencies/$agencyId/locations/type/$locationType"))
         .willReturn(aResponse()
             .withHeader("Content-type", "application/json")
-            .withBody(gson.toJson(locations))
+            .withBody(gson.toJson(locations.map { it.toMap() }))
             .withStatus(200)
         )
     )
   }
 
+  // This ignores the nullables which are not included in the JSON response
+  private fun Location.toMap(): Map<String, String> {
+    val locationMap = mutableMapOf(
+        "agencyId" to this.agencyId,
+        "currentOccupancy" to "${this.currentOccupancy}",
+        "description" to this.description,
+        "locationId" to "${this.locationId}",
+        "locationPrefix" to this.locationPrefix,
+        "locationType" to this.locationType,
+        "operationalCapacity" to "${this.operationalCapacity}",
+        "internalLocationCode" to this.internalLocationCode)
+    if (this.userDescription != null) locationMap["userDescription"] = this.userDescription as String
```

Comment:
The safe cast didn't work here but I don't know why

Reply:
Because it's calling the getter!

Reply:
Because even though you can safely call its assuming it is not null, when passed to another function its type is still nullable - hence the cast is required